include/Components/Foldout.hpp

```diff
@@ -0,0 +1,86 @@
+/*
+** Project FileShare, 2022
+**
+** Author Léo Lhuile
+**
+** Started on  Sat Jan 15 21:08:44 2022 Léo Lhuile
+** Last update Wed Sep 14 22:26:27 2022 Léo Lhuile
+**
+** Foldout.hpp : Class to manage a foldout with icon and extra informations.
+*/
+
+#pragma once
+
+#include <TGUI/TGUI.hpp>
+#include "Components/Button.hpp"
+#include "Components/List.hpp"
+
+namespace FileShare::GUI::Components {
+    class Foldout : public tgui::Group {
+        public:
+            Foldout(const char* typeName = "Components::Foldout", bool initRenderer = true);
+            ~Foldout();
+
+            typedef std::shared_ptr<Foldout> Ptr;
+			typedef std::shared_ptr<const Foldout> ConstPtr;
+
+			static Foldout::Ptr create() { return std::make_shared<Foldout>(); }
+			static Foldout::Ptr copy(Foldout::ConstPtr widget) { return widget ? std::static_pointer_cast<Foldout>(widget->clone()) : nullptr; }
+
+            tgui::Signal &getSignal(tgui::String signalName) override;
+
+            tgui::Signal onClick = { "Clicked" };
+            tgui::Signal onOpen = { "Opened" };
+            tgui::Signal onClose = { "Closed" };
+
+            void addToContent(const tgui::Widget::Ptr &item, const tgui::String &widgetName = "");
+            bool removeFromContent(const Widget::Ptr &widget);
+            const std::vector<tgui::Widget::Ptr> &getContentWidgets() const { return this->content->getWidgets(); }
```

Comment:
Same thing here : How about exposing the whole `content` object through const+non-const `getContent` functions ? It even gives access to the `content` object, not just `content->widgets`. We could also just expose the `getContentWidgets` if we dont want to expose the content object

The caller can then (if non-const) use vector functions like `push_back` (or even `emplace_back` with the performance benefits associated), and if const can call `size()` or iterate normally

Reply:
Same